Reland 2nd attempt "Add heirachical module names to torchFX graph.node"

alexmsettle opened the pull request:
Fixes #87659

Reland of PR #87742 and PR #90205

PR #90205 was reverted due to BC issues

alexmsettle committed:
Reland of commit  6b7efac due to BC issues

jerryzh168 commented on test/quantization/ao_migration/test_quantization_fx.py:
```diff
@@ -11,8 +11,6 @@ def test_function_import_quantize_fx(self):
             '_check_is_graph_module',
             '_swap_ff_with_fxff',
             '_fuse_fx',
-            'Scope',
-            'ScopeContextManager',
```
these should not be removed I think

alexmsettle replied:
Done - I added Scope and ScopeContextManager back to the test and to torch/quantization/quantize_fx.py

alexmsettle committed:
Added back Scope and ScopeContextManager symbols

to fix BC issue.

alexmsettle committed:
Added definition of Scope and ScopeContextManager

back to quantize_fx.py for backward compatibility

alexmsettle committed:
Fixed lint error

alexmsettle commented on torch/ao/quantization/quantize_fx.py:
```diff
@@ -63,59 +63,14 @@ def _fuse_fx(
-
-    """
-
+class Scope(torch.fx.proxy.Scope):
```
I added this class definition back to quantize_fx.py for BC.  It just exposes the base class implementation to this module.

Do I also need to add it to quantization/fx/tracer.py ?

jerryzh168 replied:
yeah, please add to quantization/fx/tracer.py as well, also I think you can just import the class from torch.fx.proxy instead of redefining the object?

jerryzh168 replied:
actually the definitions in this file can probably be removed, looks like I forgot to remove them in previous refactor: https://github.com/pytorch/pytorch/commit/1a7e560adecb0192f69f4d05b990800b60dc380b, I think we can just import Scope and ScopeContextManager from fx.tracer to maintain BC and remove the definitions

alexmsettle committed:
Made Scope and ScopeContextManager available

in fx/tracer.py and quantize_fx.py for BC

alexmsettle committed:
Disabled lint check F4104 for select module import

which doesn't have a use in the same file

alexmsettle committed:
Fix to BC breaking error.  Constructor signature

was changed for ScopeContextManager().  Added a wrapper class
which preserves the old signature but calls the new implementation
of the context manager.

jerryzh168 commented on torch/fx/_symbolic_trace.py:
```diff
@@ -430,9 +437,18 @@ def call_module(
-        if not self.is_leaf_module(m, module_qualified_name):
-            return forward(*args, **kwargs)
-        return self.create_proxy("call_module", module_qualified_name, args, kwargs)
+        with ScopeContextManager(self.scope, Scope(module_qualified_name, type(m))) as _scope:
```
I think we might have to check if `self` has `scope` or not, we can skip if it doesn't, to keep BC

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 commented:
note: this is no longer bc-breaking after the recent update

alexmsettle commented:
@pytorchbot merge

jerryzh168 commented:
@alexmsettle I'll merge this, looks like there are still some problems in fbcode

alexmsettle commented:
> @alexmsettle I'll merge this, looks like there are still some problems in fbcode

Do I need to stop the merge that's in progress?

jerryzh168 commented:
> > @alexmsettle I'll merge this, looks like there are still some problems in fbcode
> 
> Do I need to stop the merge that's in progress?

I have stopped merge by deleting the comment

jerryzh168 commented:
@pytorchbot rebase

jerryzh168 commented:
@pytorchbot merge
